src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLPlaceInterfaceTypeResolver.java

```diff
@@ -17,7 +20,22 @@ public GraphQLObjectType getType(TypeResolutionEnvironment environment) {
     GraphQLSchema schema = environment.getSchema();
 
     if (o instanceof BikePark) { return schema.getObjectType("BikePark"); }
-    if (o instanceof VehicleRentalStation) { return schema.getObjectType("BikeRentalStation"); }
+    if (o instanceof VehicleRentalStation) {
+      SelectionSet set = environment.getField().getFields().get(0).getSelectionSet();
+      boolean queryHasBikeRentalStationFragment = set != null && set.getSelections()
+              .stream()
+              .filter(selection -> selection instanceof InlineFragment)
```

Comment:
What about `FragmentDefinition`?

Reply:
Seems like we can only see the name of a FragmentDefinition from TypeResolutionEnvironment. So currently, we can't add support for deciding which object type to return based on those types of fragments so we will always return VehicleRentalStation. This breaks backwards compatibility a bit but we only use inline fragments for the nearest query in our UI so it's ok to have this problem for a while. I created [an issue](https://github.com/graphql-java/graphql-java/issues/2593) in graphql-java project about adding more information about fragments to the environment so this can be fixed later.